tensorflow/core/kernels/unique_op_gpu.cu.h

```diff
@@ -292,9 +292,10 @@ class UniqueOpGPU : public AsyncOpKernel {
     using namespace unique_op_gpu;
 
     // Create a fancy input iterator to indicate segment boundaries.
+    gpuprim::CountingInputIterator<TIndex> counting_iter(0);
     gpuprim::TransformInputIterator<TIndex, SegmentIndicatorFunctor<T, TIndex>,
                                     gpuprim::CountingInputIterator<TIndex>>
-        segment_indicator_iter(0, {sorted_input_ptr});
+        segment_indicator_iter(counting_iter, {sorted_input_ptr});
```

Comment:
Maybe initialize inline for readability?

Reply:
It looks like most/all of the other locations in the codebase where these iterators are used are structured like this where the counting iterator is created/initialized up front and then passed in to a transform iterator. Maybe it's best to match?